pkg/cmd/server/origin/master.go

```diff
@@ -179,7 +182,22 @@ func (c *MasterConfig) Run(protected []APIInstaller, unprotected []APIInstaller)
 	handler = c.authorizationFilter(handler)
 	handler = c.impersonationFilter(handler)
 	// audit handler must comes before the impersonationFilter to read the original user
-	handler = c.auditHandler(handler)
+	if c.Options.AuditConfig.Enabled {
```

Comment:
Kind of ugly to have this code block here. Hopefully, we are able to get rid of this in a future rebase.